pkgs/development/python-modules/craft-platforms/default.nix

```
@@ -0,0 +1,62 @@
+{
+  lib,
+  buildPythonPackage,
+  annotated-types,
+  distro,
+  fetchFromGitHub,
+  nix-update-script,
+  pytest-check,
+  pytestCheckHook,
+  pythonOlder,
+  setuptools-scm,
+}:
+
+buildPythonPackage rec {
+  pname = "craft-platforms";
+  version = "0.1.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.10";
+
+  src = fetchFromGitHub {
+    owner = "canonical";
+    repo = "craft-platforms";
+    rev = "refs/tags/${version}";
+    hash = "sha256-KzskmSw7NsH1CAYjPf2281Ob71Jd6AhWxtp5tR3IqyU=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "setuptools>=69.0" "setuptools"
+
+    substituteInPlace craft_platforms/__init__.py \
+      --replace-fail "dev" "${version}"
+  '';
+
+  build-system = [ setuptools-scm ];
```

```suggestion
  build-system = [
    setuptools
    setuptools-scm
  ];
```
Please add `setuptools` to `build-system` explicitly.

---

Done!